doc/guides/writing_tests.md

```diff
@@ -78,9 +78,11 @@ This is the body of the test. This test is quite simple, it just tests that an
 HTTP server accepts `non-ASCII` characters in the headers of an incoming
 request. Interesting things to notice:
 
-- The use of 0 as the listening port. Always use 0 instead of using an
-  arbitrary value, as it allows to run tests in parallel safely, as the
-  operating system will assign a random port.
+- If the test doesn't depend on a specific port number then always use 0 instead
+  of an arbitrary value, as it allows to run tests in parallel safely, as the
```

Comment:
`as it allows tests to be run in parallel safely` reads a little bit cleaner I think.